scripts/sync_protobuf/_utils.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+import subprocess
+import sys
+from http.client import HTTPResponse
+from pathlib import Path
+from typing import TYPE_CHECKING, Iterable
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import tomlkit
+from mypy_protobuf.main import (  # type: ignore[import-untyped]  # pyright: ignore[reportMissingTypeStubs]
+    __version__ as mypy_protobuf__version__,
+)
+
+if TYPE_CHECKING:
+    from _typeshed import StrOrBytesPath, StrPath
+
+REPO_ROOT = Path(__file__).absolute().parent.parent.parent
+MYPY_PROTOBUF_VERSION = mypy_protobuf__version__
+
+
+def download_file(url: str, destination: StrPath) -> None:
+    print(f"Downloading '{url}' to '{destination}'")
+    resp: HTTPResponse = urlopen(url)
+    if resp.getcode() != 200:
+        raise RuntimeError(f"Error downloading {url}")
+    with open(destination, "wb") as file:
+        file.write(resp.read())
```

Comment:
```suggestion
    resp: HTTPResponse
    with urlopen(url) as resp:
        if resp.getcode() != 200:
            raise RuntimeError(f"Error downloading {url}")
        with open(destination, "wb") as file:
            file.write(resp.read())
```

Reply:
Thanks, I didn't know `HTTPResponse` is a context manager